### DISCUSSION TITLE
hub name discovery

### DISCUSSION THREAD

**Discussion by Novakasa** [pybricks/support#1010] [Q&A]
Up until now, I've relied on the bleak device "name" property to find and store names for discovered hubs. This worked every time to get the name I define in the firmware install process.

```python
import asyncio

from pybricksdev.ble import find_device

async def main():

    device = await find_device()
    print(device)
    print(device.name)
    
asyncio.run(main())
```

Today, I upgraded my hubs to the latest firmware on master (+ frozen module), and this has become very unreliable. It seems that the name property now pretty much is always `None` except for some very few exceptions, where the expected name comes out. It happens so infrequently that I have not managed to reproduce it in isolation.

Output

```
1B:BC:1B:A5:81:86: None
None
```

Connecting to a device discovered using `find_device(name)` still works as expected though.

Has something changed about device discovery? What would be the recommended way right now to discover the pybricks name?

Also, I've been wondering for a while, it doesn't seem that the bluetooth address changes at all for my hubs, wasn't a random bluetooth address implemented a while ago? Even when removing the battery, the bluetooth address seems to stay the same. See: https://github.com/pybricks/support/issues/600

**Comment by Novakasa** [comment]
OK, I just downgraded bleak to 0.19.5 and now I can consistently get the hub name again, so it seems to be something with bleak 0.20.0

**Reply by dlech** [reply]
Yes, Bleak 0.20 changed how advertisement callbacks work on Windows for technical reasons. Technically, the same thing can happen on other OSes but it just happens rarely whereas on Windows it now happens most of the time. We should be able to fix this fairly easily.

**Comment by dlech** [comment]
Fixed in pybricksdev v1.0.0-alpha.41.

**Reply by Novakasa** [reply]
can confirm, works for me again 👍

**Comment by dlech** [comment]
> Also, I've been wondering for a while, it doesn't seem that the bluetooth address changes at all for my hubs, wasn't a random bluetooth address implemented a while ago? Even when removing the battery, the bluetooth address seems to stay the same.

Which hub and which firmware version?

**Reply by Novakasa** [reply]
Both Technic and City hub, firmware version current master https://github.com/pybricks/pybricks-micropython/commit/415910bd5f53cb538f4b58c8a23d2030096c86b6 + frozen module

**Reply by Novakasa** [reply]
I think I've also wondered about this with 3.2.1 stable
Edit: Not quite stable, it was this commit + frozen module: https://github.com/pybricks/pybricks-micropython/commit/030d06823d2386262f568174a78a1f6032d8b85f
